homeassistant/components/evohome/__init__.py

```diff
@@ -58,16 +68,57 @@
     extra=vol.ALLOW_EXTRA,
 )
 
+ATTR_SYSTEM_MODE = "mode"
+ATTR_DURATION_DAYS = "days"
+ATTR_DURATION_HOURS = "hours"
+
+ATTR_ZONE_TEMP = "setpoint"
+ATTR_UNTIL_MINUTES = "duration"
+ATTR_UNTIL_TIME = "until"
+
+SVC_REFRESH_SYSTEM = "force_refresh"
+SVC_SET_SYSTEM_MODE = "set_system_mode"
+SVC_RESET_SYSTEM = "reset_system"
+SVC_SET_ZONE_MODE = "set_zone_override"
+SVC_RESET_ZONE_MODE = "clear_zone_override"
+
+HH_MM_REGEXP = r"^(0[0-9]|1[0-9]|2[0-3]):[0-5][0-9]$"  # 24h format, with leading 0
+
+SET_SYSTEM_MODE_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_SYSTEM_MODE): vol.In(EVO_SYSTEM_MODES),
+        vol.Exclusive(ATTR_DURATION_DAYS, "duration"): vol.All(
+            cv.positive_int, vol.Range(min=1, max=99)
+        ),
+        vol.Exclusive(ATTR_DURATION_HOURS, "duration"): vol.All(
+            cv.positive_int, vol.Range(min=1, max=24)
+        ),
+    }
+)
+SET_ZONE_MODE_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+        vol.Required(ATTR_ZONE_TEMP): vol.Coerce(float),
+        vol.Exclusive(ATTR_UNTIL_MINUTES, "until"): vol.All(
```

Comment:
Is there a reason we don't use a time interval here too?

Reply:
See above.